azure/k8s-nginx-lb-script.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# set -o errexit
+# set -o pipefail
+# set -o nounset
+
+source ../scripts/check-available-commands.sh
+checkCommandsAvailable helm az kubectl
+
+export RESOURCE_GROUP="$(terraform output -raw resource_group)"
+export CLUSTER_NAME="$(terraform output -raw cluster_name)"
+
+
+echo "setting up kubectl"
+
+az aks get-credentials --resource-group $RESOURCE_GROUP --name $CLUSTER_NAME
+
+echo "Installing the nginx ingress controller chart"
+
+helm upgrade --install ingress-nginx ingress-nginx --repo https://kubernetes.github.io/ingress-nginx --namespace ingress-nginx --create-namespace
+
+
+echo "apply -f k8s/wrongsecrets-balancer-service.yml in 10 s"
+sleep 10
+kubectl apply -f k8s/wrongsecrets-balancer-service.yml
+
+echo "apply -f k8s/wrongsecrets-balancer-ingress.yml in 10 s"
+sleep 10
+kubectl apply -f k8s/wrongsecrets-balancer-ingress.yml
+
+kubectl apply -f k8s/ctfd-service.yaml
+kubectl apply -f k8s/ctfd-ingress.yaml
+
+IP_ADDRESS="$(kubectl get service ingress-nginx-controller --namespace=ingress-nginx --output jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+
+echo "IP_ADDRESS for the entrypoint is $IP_ADDRESS"
+
+echo "You can now access the Wrongsecrets ingress at http://$IP_ADDRESS"
+
+echo "You can now access the ctfd ingress at http://$IP_ADDRESS/ctfd"
```

Comment:
<img width="413" alt="image" src="https://github.com/OWASP/wrongsecrets-ctf-party/assets/1457214/5098ac17-b966-4aec-a2e8-385c088e9f67">
something seems to go wrong when trying to access it.

Reply:
As we discussed, the helm chart for the ctfd appeared to need a root path as all redirected requests are not considering the prefix path (conflicting with our balancer ingress which has the same requirement and we only have one IP for the entry-point). Since we do not want to force users to have a domain name in the terraform configuration. I edited this file to advise users to access the ctfd ingress through kubectl port forwarding.